Sources/PointFree/RequestContext.swift

```diff
@@ -42,47 +42,36 @@ public func requireUser<A>(
   -> Middleware<StatusLineOpen, ResponseEnded, A, Data> {
 
     return { conn in
-
-      let currentUser = extractedGitHubUserEnvelope(from: conn.request)
-        .map(fetchDatabaseUser)
-        ?? pure(nil)
-
-      return currentUser.flatMap { user in
-        user.map { conn.map(const($0 .*. conn.data)) |> middleware }
-          ?? (conn |> redirect(to: path(to: .login(redirect: conn.request.url?.absoluteString))))
+      (conn |> readSessionCookieMiddleware)
+        .flatMap { c in
+          c.data.first.map { user in
+            c.map(const(user .*. c.data.second))
+              |> middleware
+            }
+            ?? (conn |> redirect(to: .login(redirect: conn.request.url?.absoluteString)))
       }
     }
 }
 
-//func currentUserMiddleware<A, I>(
-//  _ conn: Conn<I, A>
-//  ) -> IO<Conn<I, T2<Database.User?, A>>> {
-//
-//  let currentUser = extractedGitHubUserEnvelope(from: conn.request)
-//    .map(fetchDatabaseUser)
-//    ?? pure(nil)
+//public func _requireUser<A>(
+//  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data>)
+//  -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User?, A>, Data> {
 //
-//  return currentUser.map { user in
-//    conn.map(
-//      const(user .*. conn.data)
-//    )
-//  }
+//    return { conn in
+//      conn.data.first.map { user in
+//        conn.map(const(T2(first: user, second: conn.data.second)))
+//        }
+//        .map(middleware)
+//        ?? (conn |> redirect(to: .login(redirect: conn.request.url?.absoluteString)))
+//    }
 //}
 
-// todo: maybe do this, maybe not
-//public func .*. <A, B> (lhs: A, rhs: B) -> Tuple<A, B> {
-//  return .init(first: lhs, second: rhs)
-//}
+func currentUserMiddleware<A, I>(
+  _ conn: Conn<I, A>
+  ) -> IO<Conn<I, Tuple2<Database.User?, A>>> {
 
-//func currentRequestMiddleware<A, I>(
-//  _ conn: Conn<I, A>
-//  ) -> IO<Conn<I, T2<URLRequest, A>>> {
-//
-//  return pure <|
-//    conn.map(
-//      const(conn.request .*. conn.data)
-//  )
-//}
+  return conn |> readSessionCookieMiddleware
```

Comment:
we can just get rid of `currentUserMiddleware` cause it's the same as `readSessionCookieMiddleware`. might want a better name.